crates/edr_rpc_server/src/filter.rs

```diff
@@ -1,82 +1,40 @@
 use std::time::{Duration, Instant};
 
-use edr_eth::{
-    remote::{
-        filter::{FilterBlockTarget, FilterOptions, FilteredEvents, OneOrMoreAddresses},
-        jsonrpc::ResponseData,
-        BlockSpec,
-    },
-    Address, B256, U256,
-};
-use edr_evm::{
-    blockchain::{BlockchainError, SyncBlockchain},
-    state::StateError,
-};
+use edr_eth::remote::filter::{FilteredEvents, LogOutput};
 
-use super::{_block_number_from_block_spec, _block_number_from_hash};
-
-pub struct _FilterCriteria {
```

Comment:
What happened to the filter criteria and the places it was used?

Reply:
It wasn't used before (hence the underscore prefix I think)

Reply:
It was in the `struct Filter`:
![image](https://github.com/NomicFoundation/hardhat/assets/6917585/585e3ce5-5174-4ab5-a79a-4c9593b7477a)

Reply:
If I'm not mistaken, those get used by the functions in: https://github.com/NomicFoundation/edr/issues/114.

Reply:
Ah sorry I thought you asked if functionality was lost